nifi-nar-bundles/nifi-druid-bundle/nifi-druid-controller-service/src/main/java/org/apache/nifi/controller/DruidTranquilityController.java

```diff
@@ -0,0 +1,416 @@
+		        
+		final Beam<Map<String, Object>> beam = DruidBeams.builder(timestamper)
+			    		.curator(curator)
+		                .discoveryPath(discoveryPath)
+		                .location(DruidLocation.create(DruidEnvironment.create(indexService, firehosePattern),dataSource))
+		                .timestampSpec(timestampSpec)
+		                .rollup(DruidRollup.create(DruidDimensions.specific(dimensions), aggregator, QueryGranularity.fromString(queryGranularity)))
+		                .tuning(
+		                        ClusteredBeamTuning
+		                                .builder()
+		                                //.segmentGranularity(Granularity.MINUTE)
+		                                .segmentGranularity(getSegmentGranularity(segmentGranularity))
+		                                .windowPeriod(new Period(windowPeriod))
+		                                .partitions(clusterPartitions)
+		                                .replicants(clusterReplication)
+		                                .build()
+		                )
+		                .druidBeamConfig(
+		                        DruidBeamConfig
+		                                .builder()
+		                                .indexRetryPeriod(new Period(indexRetryPeriod))
+		                                .build())
+		                .buildBeam();
+		
+		tranquilizer = Tranquilizer.builder()
+                .maxBatchSize(10000000)
+                .maxPendingBatches(1000)
+                .lingerMillis(1000)
+                .blockOnFull(true)
+                .build(beam);
+		
+        tranquilizer.start();
+	}
+	
+	public Tranquilizer getTranquilizer(){
+		return tranquilizer;
+	}
+	
+	private  List<Map<String, String>> parseJsonString(String aggregatorJson) {
+        ObjectMapper mapper = new ObjectMapper();
+        List<Map<String, String>> aggSpecList = null;
+        try {
+        	getLogger().debug("Druid Tranquility Service: Aggregator Spec as String: " + aggregatorJson);
+            aggSpecList = mapper.readValue(aggregatorJson, List.class);
+            getLogger().debug("Druid Tranquility Service: Aggregator Spec as List: " + aggSpecList);
+        	return aggSpecList;
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Exception while parsing the aggregratorJson");
+        }
+    }
+	
+	private List<String> getDimensions(String dimensionStringList){
+		List<String> dimensionList = new LinkedList(Arrays.asList(dimensionStringList.split(",")));
+		return dimensionList;
+	}
+	
+	private List<AggregatorFactory> getAggregatorList(String aggregatorJSON) {
+        List<AggregatorFactory> aggregatorList = new LinkedList<>();
+        List<Map<String, String>> aggregatorInfo = parseJsonString(aggregatorJSON);
+        for (Map<String, String> aggregator : aggregatorInfo) {
+
+            if (aggregator.get("type").equalsIgnoreCase("count")) {
+                //Map<String, String> map = aggregator.get("count");
```

Comment:
recommend killing off the dead/commented code

Reply:
removed dead comments.